### ISSUE TITLE
Bump Scala version to 3.3.0+ to avoid lazy val not working properly under GraalVM Native Image

### ISSUE THREAD

**Issue by WtzLAS** [kevin-lee/just-semver#186]
According to [this comment](https://github.com/lampepfl/dotty/issues/13985#issuecomment-1318525950),  existing libraries need to bump a version to be working properly under GraalVM Native Image. :(

```
timestamp=2023-10-01T04:48:34.846238+08:00 level=ERROR thread=zio-fiber-1 message="" cause=Exception in thread "zio-fiber-4" java.lang.ExceptionInInitializerError: null
        at java.base@21/java.lang.Class.ensureInitialized(DynamicHub.java:595)
        at just.semver.expr.ComparisonOperator$.$new(ComparisonOperator.scala:14)
        at just.semver.expr.ComparisonOperator$.<clinit>(ComparisonOperator.scala:9)
        at just.semver.matcher.SemVerComparison$.parse$$anonfun$2(SemVerComparison.scala:21)
        at scala.util.Either.flatMap(Either.scala:352)
        at just.semver.matcher.SemVerComparison$.parse(SemVerComparison.scala:38)
        at just.semver.matcher.SemVerMatchers$.$anonfun$12(SemVerMatchers.scala:104)
        at scala.collection.ArrayOps$.map$extension(ArrayOps.scala:932)
        at just.semver.matcher.SemVerMatchers$.each$8(SemVerMatchers.scala:108)
        at just.semver.matcher.SemVerMatchers$.$anonfun$2(SemVerMatchers.scala:128)
        at scala.collection.ArrayOps$.map$extension(ArrayOps.scala:932)
        at just.semver.matcher.SemVerMatchers$.parse(SemVerMatchers.scala:128)
        at just.semver.matcher.SemVerMatchers$.unsafeParse(SemVerMatchers.scala:144)
        at routes.Routes$package$.<clinit>(Routes.scala:33)
        at Main$.run(Main.scala:10)
        at zio.ZIOAppPlatformSpecific.$anonfun$3$$anonfun$1$$anonfun$1(ZIOAppPlatformSpecific.scala:23)
        at zio.Scope$ExtendPartiallyApplied$.apply$extension(Scope.scala:181)
        at zio.Scope$UsePartiallyApplied$.apply$extension(Scope.scala:186)
        at zio.ZIO$ScopedPartiallyApplied$.apply$extension$$anonfun$7(ZIO.scala:5342)
        Suppressed: java.lang.NoSuchFieldException: 0bitmap$1
                at java.base@21/java.lang.Class.checkField(DynamicHub.java:1031)
                at java.base@21/java.lang.Class.getDeclaredField(DynamicHub.java:1146)
                at scala.runtime.LazyVals$.getOffset(LazyVals.scala:140)
                at just.semver.expr.ComparisonOperator.<clinit>(ComparisonOperator.scala:17)
                at java.base@21/java.lang.Class.ensureInitialized(DynamicHub.java:595)
                at just.semver.expr.ComparisonOperator$.$new(ComparisonOperator.scala:14)
                at just.semver.expr.ComparisonOperator$.<clinit>(ComparisonOperator.scala:9)
                at just.semver.matcher.SemVerComparison$.parse$$anonfun$2(SemVerComparison.scala:21)
                at scala.util.Either.flatMap(Either.scala:352)
                at just.semver.matcher.SemVerComparison$.parse(SemVerComparison.scala:38)
                at just.semver.matcher.SemVerMatchers$.$anonfun$12(SemVerMatchers.scala:104)
                at scala.collection.ArrayOps$.map$extension(ArrayOps.scala:932)
                at just.semver.matcher.SemVerMatchers$.each$8(SemVerMatchers.scala:108)
                at just.semver.matcher.SemVerMatchers$.$anonfun$2(SemVerMatchers.scala:128)
                at scala.collection.ArrayOps$.map$extension(ArrayOps.scala:932)
                at just.semver.matcher.SemVerMatchers$.parse(SemVerMatchers.scala:128)
                at just.semver.matcher.SemVerMatchers$.unsafeParse(SemVerMatchers.scala:144)
                at routes.Routes$package$.<clinit>(Routes.scala:33)
                at Main$.run(Main.scala:10)
                at zio.ZIOAppPlatformSpecific.$anonfun$3$$anonfun$1$$anonfun$1(ZIOAppPlatformSpecific.scala:23)
                at zio.Scope$ExtendPartiallyApplied$.apply$extension(Scope.scala:181)
                at zio.Scope$UsePartiallyApplied$.apply$extension(Scope.scala:186)
                at zio.ZIO$ScopedPartiallyApplied$.apply$extension$$anonfun$7(ZIO.scala:5342)
```

**Comment by kevin-lee** [comment]
@WtzLAS Thank you! I'll sort it out shortly.

**Comment by kevin-lee** [comment]
There will be more than one release so please be patient.

**Comment by WtzLAS** [comment]
As a temporary solution, this is a working `reflect-config.json`:

**Comment by kevin-lee** [comment]
Fixed by #199. It will be released soon.

**Comment by kevin-lee** [comment]
<details>
  <summary>Click to expand</summary>

> As a temporary solution, this is a working `reflect-config.json`:
> 
> ```json
> [
>   {
>     "name": "just.semver.expr.ComparisonOperator$",
>     "fields": [
>       {
>         "name": "0bitmap$1"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.SemVer$Major$",
>     "fields": [
>       {
>         "name": "0bitmap$1"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.SemVer$Minor$",
>     "fields": [
>       {
>         "name": "0bitmap$2"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.SemVer$Patch$",
>     "fields": [
>       {
>         "name": "0bitmap$3"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.SemVer$",
>     "fields": [
>       {
>         "name": "0bitmap$4"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.matcher.SemVerComparison$",
>     "fields": [
>       {
>         "name": "0bitmap$2"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.matcher.SemVerMatcher$",
>     "fields": [
>       {
>         "name": "0bitmap$2"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.matcher.SemVerMatchers$And$",
>     "fields": [
>       {
>         "name": "0bitmap$2"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.matcher.SemVerMatchers$Or$",
>     "fields": [
>       {
>         "name": "0bitmap$1"
>       }
>     ]
>   },
>   {
>     "name": "just.semver.matcher.SemVerMatchers$",
>     "fields": [
>       {
>         "name": "0bitmap$3"
>       }
>     ]
>   }
> ]
> ```
</details>
@WtzLAS Thank you for the additional information above.
I've just released `0.11.0`, which was compiled with Scala `3.3.0`, so please test it and let me know the result.

**Comment by WtzLAS** [comment]
> Click to expand
> @WtzLAS Thank you for the additional information above. I've just released `0.11.0`, which was compiled with Scala `3.3.0`, so please test it and let me know the result.

confirmed working with GraalVM Native Image, thank you!

**Comment by kevin-lee** [comment]
> > @WtzLAS Thank you for the additional information above. I've just released `0.11.0`, which was compiled with Scala `3.3.0`, so please test it and let me know the result.
> 
> confirmed working with GraalVM Native Image, thank you!

@WtzLAS That's great! Thanks for the confirmation.
